game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -1202,6 +1162,86 @@ && canAirSafelyLandAfterAttack(unit, t)) {
     }
   }
 
+  private void determineTerritoriesThatCanBeBombed(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Unit> alreadyAttackedWithUnits) {
+    final Map<Unit, Set<Territory>> bomberMoveMap =
+        territoryManager.getAttackOptions().getBomberMoveMap();
+    final Set<Unit> bombers = bomberMoveMap.keySet();
+    bombers.removeAll(alreadyAttackedWithUnits);
+    final boolean raidsMayBePrecededByAirBattles =
+        Properties.getRaidsMayBePreceededByAirBattles(data.getProperties());
+    for (final Unit bomber : bombers) {
+      final Set<Territory> bomberTargetTerritories = bomberMoveMap.get(bomber);
+      bomberTargetTerritories.removeIf(
+          terr ->
+              raidsMayBePrecededByAirBattles
+                  && AirBattle.territoryCouldPossiblyHaveAirBattleDefenders(
+                      terr, player, data, true));
+      determineBestBombingAttackForBomber(
+          attackMap, sortedUnitAttackOptions, bomberTargetTerritories, bomber);
+    }
+  }
+
+  /**
+   * Determines the best bombing attack territory for a single {@code bomber}.
+   *
+   * @param attackMap Attack map
+   * @param sortedUnitAttackOptions Sorted attack options
+   * @param bomberTargetTerritories Target territories
+   * @param bomber Bomber unit
+   */
+  private void determineBestBombingAttackForBomber(
+      final Map<Territory, ProTerritory> attackMap,
+      final Map<Unit, Set<Territory>> sortedUnitAttackOptions,
+      final Set<Territory> bomberTargetTerritories,
+      final Unit bomber) {
+    final Predicate<Unit> bombingTargetMatch =
+        Matches.unitCanProduceUnitsAndCanBeDamaged()
+            .and(Matches.unitIsLegalBombingTargetBy(bomber));
+    Optional<Territory> maxBombingTerritory = Optional.empty();
+    int maxBombingScore = MIN_BOMBING_SCORE;
+    for (final Territory t : bomberTargetTerritories) {
+      final List<Unit> targetUnits = t.getUnitCollection().getMatches(bombingTargetMatch);
+      if (!targetUnits.isEmpty() && canAirSafelyLandAfterAttack(bomber, t)) {
+        final int noAaBombingDefense =
+            t.anyUnitsMatch(Matches.unitIsAaForBombingThisUnitOnly()) ? 0 : 1;
+        int maxDamageProduction = 0; // territory damage + target unit damage(s)
```

Comment:
I find the name a bit confusing. What do you think about `maxPossibleDamage` instead?

Also, I don't understand the comment after it, since it's implying it's adding stuff up, but this is just set based on the territory attachment.

Reply:
Removed the comment.
The name `maxDamageProduction` is more suited from my perspective as it is the maximum damage to the territory production and the word "damage" is otherwise used to refer to units.